ReSharper.FSharp/src/FSharp.Psi.Daemon/src/Analyzers/LambdaAnalyzer.fs

```diff
@@ -185,6 +186,44 @@ type LambdaAnalyzer() =
             | _ -> ctor arg
         | _ -> ctor arg
 
+    let rec containsForcedCalculations (expression: IFSharpExpression) =
+        let mutable containsForcedCalculations = false
+        let mutable prefixAppExprContext: IPrefixAppExpr = null
+        let processor = { new IRecursiveElementProcessor with
+            member x.ProcessingIsFinished = containsForcedCalculations
+            member x.InteriorShouldBeProcessed(treeNode) = not (treeNode :? ILambdaExpr)
+            member x.ProcessAfterInterior(treeNode) = ()
+            member x.ProcessBeforeInterior(treeNode) =
+                match treeNode with
+                | :? INewExpr
+                | :? IForLikeExpr
+                | :? IIndexerExpr
+                | :? IBinaryAppExpr ->
+                    containsForcedCalculations <- true
+                | :? IPrefixAppExpr as prefixAppExpr ->
+                    if prefixAppExpr.IsIndexerLike then containsForcedCalculations <- true else
+                    if isNull (PrefixAppExprNavigator.GetByFunctionExpression(prefixAppExpr)) then
+                        prefixAppExprContext <- prefixAppExpr
+                | :? IReferenceExpr as referenceExpr ->
+                    containsForcedCalculations <-
+                        match referenceExpr.Reference.GetFcsSymbol() with
+                        | :? FSharpMemberOrFunctionOrValue as m ->
+                            m.IsProperty || m.IsTypeFunction || m.IsMutable ||
+                            isNotNull prefixAppExprContext &&
+                            (m.IsConstructor ||
+                            (m.IsFunction || m.IsMethod) &&
```

Comment:
Is this check really needed or the parameter group count check would be enough?
Does it cover active patterns and operators, like `(+) 1` and `(|A|) 1`?

Reply:
> Does it cover active patterns and operators, like (+) 1 and (|A|) 1?

Yes, because `IsFunction` = true for them


> Is this check really needed or the parameter group count check would be enough?

I think the parameter group count check would be enough, yes

Reply:
Or not enough, because `list[0]` satisfies the check, but but we want to allow such a case
https://github.com/JetBrains/resharper-fsharp/pull/437#discussion_r1010453025